include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -84,20 +91,34 @@ struct FFT<2,false, Scalar, device_capabilities>
         complex_t<Scalar> lo = {loVec.x, loVec.y};  
         complex_t<Scalar> hi = {hiVec.x, hiVec.y};
 
-        // special first iteration
-        if (_NBL_HLSL_WORKGROUP_SIZE_ > glsl::gl_SubgroupSize())
+        // special first iteration - only if workgroupsize > subgroupsize
+        if (_NBL_HLSL_WORKGROUP_SIZE_ ^ glsl::gl_SubgroupSize())
             fft::DIF<Scalar>::radix2(fft::twiddle<false, Scalar>(threadID, _NBL_HLSL_WORKGROUP_SIZE_), lo, hi); 
 
         // Run bigger steps until Subgroup-sized
         for (uint32_t stride = _NBL_HLSL_WORKGROUP_SIZE_ >> 1; stride > glsl::gl_SubgroupSize(); stride >>= 1)
+        {   
+            // If at least one loop was executed, we must wait for all threads to get their values before we write to shared mem again
+            if ( !(stride & (_NBL_HLSL_WORKGROUP_SIZE_ >> 1)) )
+                sharedmemAdaptor.workgroupExecutionAndMemoryBarrier(); 
```

Comment:
put the barrier AFTER the `FFT_loop` call, you don't need to barrier before because you always start with a "fresh, uncontented" accessor